src/components/App/App.js

```diff
@@ -13,35 +14,53 @@ class App extends Component {
     super()
     this.state = {
       teas: [],
-      error: false
+      error: ''
     }
   }
 
+  returnHome = () => {
+    this.setState({ teas: [], error: ''})
+  }
+
   componentDidMount = () => {
     getTeas()
       .then(data => this.setState({ teas: data }))
-      .catch(error => this.setState({ error: error.message }))
+      .catch(error => this.setState({ error: `${error.message}. We're busy taking care of a kettle that's boiling over! Please visit us again later.` }))
   }
 
   render() {
     return (
       <main className='App'>
         <Header />
-        <Switch>
-          <Route 
-            exact path='/:id'
-            render={({ match }) => {
-              const teaId = match.params.id
-              return <TeaPage selectedTea={teaId}/>
-            }}
-          />
-          <Route 
-            exact path='/' 
-            render={() => {
-              return <TeaContainer teas={this.state.teas}/>
-            }}
-          />
-        </Switch>
+        <div className='components-container'>
+          {!this.state.teas.length 
+          ? <Error 
+            errorMessage={this.state.error} 
+            returnHome={this.returnHome} />
+          : <Switch>
+            <Route 
+              exact path='/' 
+              render={() => {
+                return <TeaContainer 
+                teas={this.state.teas} 
+                returnHome={this.returnHome} />
+              }}
+            />
+            <Route 
+              exact path='/:id'
+              render={({ match }) => {
+                const teaId = match.params.id
+                return <TeaPage 
+                  teas={this.state.teas} 
+                  selectedTea={parseInt(teaId)} 
+                  returnHome={this.returnHome}
+                  errorMessage={this.state.error} />
```

Comment:
this whole `match` part could be moved to `<TeaPage>` component. no need to figure out id at this level